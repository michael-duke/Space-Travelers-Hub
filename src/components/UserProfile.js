import React from 'react';
import { useSelector } from 'react-redux';
import { Button } from '@material-tailwind/react';
import { allMissions } from '../redux/missions/missionsSlice';
import { allRockets } from '../redux/rockets/rocketsSlice';

const UserProfile = () => {
  const missions = useSelector(allMissions);
  const rockets = useSelector(allRockets);

  const missionsJoined = missions.filter((mission) => mission.joined);
  const rocketsReserved = rockets.filter((rocket) => rocket.rocketReservation);

  return (
    <div className="flex min-h-screen items-center gap-8 mx-10">
      <div className="missions w-1/2 text-left">
        <h2 className="text-2xl">My Missions</h2>
        {missionsJoined.length === 0
          ? (
            <>
              <span className="italic">No Missions Joined, yet</span>
              <span> ðŸŽ¯ðŸ˜Š. </span>
              <span className="italic">Click on Missions page to join.</span>
            </>
          )
          : missionsJoined.map(({ missionId: id, missionName }) => (
            <div key={id} className="border text-left p-5 flex items-center justify-between">
              {missionName}
              <Button>
                Leave
              </Button>
            </div>
          ))}
      </div>
      <div className="rockets w-1/2 text-left">
        <h2 className="text-2xl">My Rockets</h2>
        {rocketsReserved.length === 0
          ? (
            <>
              <span className="italic">No Rockets Reserved, yet</span>
              <span> ðŸš€ðŸ˜Š. </span>
              <span className="italic">Click on Rockets page to reserve.</span>
            </>
          )
          : rocketsReserved.map(({ missionId: id, missionName }) => (
            <div key={id} className="border text-left p-5 flex items-center justify-between">
              {missionName}
              <Button>
                Cancel
              </Button>
            </div>
          ))}
      </div>
    </div>
  );
};
export default UserProfile;
