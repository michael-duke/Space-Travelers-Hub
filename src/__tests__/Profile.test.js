import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import UserProfile from '../components/UserProfile';

describe('Test for UserProfile component', () => {
  test('Renders UserProfile correctly', () => {
    const userProfile = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <UserProfile />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(userProfile).toMatchSnapshot();
  });

  test('Check for heading \'My Missions\' on the UserProfile component is in the rendered document', () => {
    render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <UserProfile />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(
      screen.getByRole('heading', { name: /My Missions/i }),
    ).toBeInTheDocument();
  });

  test('Check for heading \'My Rockets\' on the UserProfile component is in the rendered document', () => {
    render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <UserProfile />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(
      screen.getByRole('heading', { name: /My Rockets/i }),
    ).toBeInTheDocument();
  });
});
