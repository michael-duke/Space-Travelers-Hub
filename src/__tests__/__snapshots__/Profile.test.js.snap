// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for UserProfile component Renders UserProfile correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=" py-20 flex sl:flex-col sl:justify-center sl:items-stretch min-h-screen items-center gap-8 mx-10 sm:mx-2"
      >
        <div
          class="missions w-1/2 sm:w-2/3 sl:w-full text-left mt-8 max-h-96 overflow-y-scroll"
        >
          <h2
            class="text-2xl text-center"
          >
            My Missions
          </h2>
          <span
            class="italic"
          >
            No Missions Joined, yet
          </span>
          <span>
             ðŸŽ¯ðŸ˜Š. 
          </span>
          <span
            class="italic"
          >
            Click on Missions page to join.
          </span>
        </div>
        <div
          class="rockets w-1/2 sm:w-2/3 sl:w-full text-left mt-8 max-h-96 overflow-y-scroll"
        >
          <h2
            class="text-2xl text-center"
          >
            My Rockets
          </h2>
          <span
            class="italic"
          >
            No Rockets Reserved, yet
          </span>
          <span>
             ðŸš€ðŸ˜Š. 
          </span>
          <span
            class="italic"
          >
            Click on Rockets page to reserve.
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" py-20 flex sl:flex-col sl:justify-center sl:items-stretch min-h-screen items-center gap-8 mx-10 sm:mx-2"
    >
      <div
        class="missions w-1/2 sm:w-2/3 sl:w-full text-left mt-8 max-h-96 overflow-y-scroll"
      >
        <h2
          class="text-2xl text-center"
        >
          My Missions
        </h2>
        <span
          class="italic"
        >
          No Missions Joined, yet
        </span>
        <span>
           ðŸŽ¯ðŸ˜Š. 
        </span>
        <span
          class="italic"
        >
          Click on Missions page to join.
        </span>
      </div>
      <div
        class="rockets w-1/2 sm:w-2/3 sl:w-full text-left mt-8 max-h-96 overflow-y-scroll"
      >
        <h2
          class="text-2xl text-center"
        >
          My Rockets
        </h2>
        <span
          class="italic"
        >
          No Rockets Reserved, yet
        </span>
        <span>
           ðŸš€ðŸ˜Š. 
        </span>
        <span
          class="italic"
        >
          Click on Rockets page to reserve.
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
